### PR TITLE
primary branch can be main

### PR DESCRIPTION
**Pull request by tkatzir**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39616

### CODE REVIEW

**Review by dorschw — COMMENTED** [review]
Looks good

File: demisto_sdk/commands/common/git_util.py [hunk 1]
```diff
@@ -422,21 +422,30 @@ def check_if_remote_exists(self, remote):
...
         # check for sha1 in regex
         sha1_pattern = re.compile(r'\b[0-9a-f]{40}\b', flags=re.IGNORECASE)
-        if sha1_pattern.match(prev_ver):
+        if prev_ver and sha1_pattern.match(prev_ver):
             return None, prev_ver
 
-        if '/' in prev_ver:
-            remote = prev_ver.split('/')[0]
+        if prev_ver and '/' in prev_ver:
+            prev_ver_split = prev_ver.split('/')
```
Comment (yaakovi) [inline demisto_sdk/commands/common/git_util.py:432]: then you can remove lines 433, 435
```suggestion
            remote, branch = prev_ver.split('/', 1)
```

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py [hunk 2]
```diff
@@ -101,7 +103,10 @@ def is_release_branch():
         Returns:
             (bool): is release branch
         """
-        diff_string_config_yml = run_command("git diff origin/master .circleci/config.yml")
+        git_util = GitUtil(repo=Content.git())
+        main_branch = git_util.handle_prev_ver()[1]
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/content_entity_validator.py:107]: should this be a function in `tools.py` or `git_util.py`?

File: demisto_sdk/commands/secrets/secrets.py [hunk 3]
```diff
@@ -79,7 +81,8 @@ def __init__(
         self.is_circle = is_circle
         self.white_list_path = white_list_path
         self.ignore_entropy = ignore_entropy
-        self.prev_ver = prev_ver if prev_ver is not None else 'origin/master'
+        # self.prev_ver = prev_ver if prev_ver is not None else 'origin/master'
```
Comment (yaakovi) [inline demisto_sdk/commands/secrets/secrets.py:84]: remove if not relevant

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 4]
```diff
@@ -49,6 +51,8 @@ def __init__(self, pack_path: str, update_type: Union[str, None], modified_files
         self.should_delete_existing_rn = False
         self.pack_metadata_only = pack_metadata_only
         self.is_force = is_force
+        self.git_util = GitUtil(repo=Content.git())
```
Comment (yaakovi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:54]: why do you need to save the `git_util` object? it's not being used elsewhere

File: demisto_sdk/commands/validate/validate_manager.py [hunk 5]
```diff
@@ -1224,11 +1224,17 @@ def setup_prev_ver(self, prev_ver: Optional[str]):
         if prev_ver:
             return prev_ver
 
-        # check if git is connected and if demisto exists in remotes if so set prev_ver as 'demisto/master'
-        if self.git_util and self.git_util.check_if_remote_exists('demisto'):
-            return 'demisto/master'
+        # If git is connected - Use it to get prev_ver
+        if self.git_util:
+            # If demisto exists in remotes if so set prev_ver as 'demisto/master'
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/validate_manager.py:1229]:
```suggestion
            # If demisto exists in remotes - set prev_ver as 'demisto/master'
```

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py (hunk 2 above)
Comment (tkatzir) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py:107]: It's relevant when git is used, so I say keep it in git_util.py

File: demisto_sdk/commands/update_release_notes/update_rn.py (hunk 4 above)
Comment (tkatzir) [reply demisto_sdk/commands/update_release_notes/update_rn.py]: https://github.com/demisto/demisto-sdk/pull/1758/commits/9f722bc6cf8be951ecad4cba2c801dae1d1b0cef

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py (hunk 2 above)
Comment (yaakovi) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py:107]: ok, then let's move it to `git_util.py`

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 6]
```diff
@@ -45,10 +45,13 @@ def __init__(self, file_path: str, known_words_file_path: str = None, no_camel_c
...
         self.file_path = file_path
         self.git_util = None
-        self.prev_ver = prev_ver if prev_ver else 'demisto/master'
 
         if use_git:
             self.git_util = GitUtil()
+            self.git_util = GitUtil(repo=Content.git())
+            self.prev_ver = self.git_util.handle_prev_ver()[1]
+        else:
+            self.prev_ver = prev_ver if prev_ver else 'demisto/master'
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply demisto_sdk/commands/doc_reviewer/doc_reviewer.py:53]: git_util is only used when git is used.

File: demisto_sdk/commands/secrets/secrets.py [hunk 7]
```diff
@@ -128,11 +131,15 @@ def get_all_diff_text_files(self, branch_name, is_circle):
...
-                self.prev_ver = 'origin/' + self.prev_ver
-            print(f"Running secrets validation against {self.prev_ver}")
-
-            changed_files_string = run_command(f"git diff --name-status {self.prev_ver}...{branch_name}")
+            prev_ver = self.prev_ver
+            if not prev_ver:
+                self.git_util = GitUtil(repo=Content.git())
+                prev_ver = self.git_util.handle_prev_ver()[1]
+            if not prev_ver.startswith('origin'):
+                prev_ver = 'origin/' + prev_ver
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply demisto_sdk/commands/secrets/secrets.py:138]: git_util is only used when git is used.

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py (hunk 2 above)
Comment (tkatzir) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py:107]: Do you mean like a static method?

**Review by dorschw — APPROVED** [review]

**Pull request merged**